_docs-sources/pipelines/architecture/index.md

```diff
@@ -0,0 +1,45 @@
+# Architecture
+
+## Overview
+
+There are three major components in the Gruntwork Pipelines architecture:
+
+1. The orchestrator
+2. The dispatcher
+3. The executor
+
+### The orchestrator
+
+The orchestrator identifies each [infrastructure change](../core-concepts/#infrastructure-change) in the Pull Request or git commit, classifies the type of infrastructure change it is (e.g. NewAwsAccount vs. OpenTfModuleUpdate) and determines the right [pipelines actions](../core-concepts/#pipelines-actions) to run based on that infrastructure change. For each infrastructure change, it calls the dispatcher.
+
+### The dispatcher
+
+The dispatcher takes as input an infrastructure change and a list of which pipelines actions to run with along with what parameters, and calls the executor to run these actions on the right infrastructure change.
+
+### The executor
+
+The executor executes an ordered list of pipelines actions against a specific pull request or git commit in response to a specific infrastructure. It then posts the status and output of these pipelines actions to the pull request, or takes other actions to ensure that the user has visibility into what happened.
```

Comment:
It doesn't respect the ordered list right now. We deferred that work in order to better support parallelization sooner.

Reply:
Roger, please update as needed